ui/src/components/app-form/app-form.tsx

```diff
@@ -62,6 +65,8 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
   const firstErrorRef = useRef<HTMLInputElement | null>(null);
   const appInfoRef = useRef<HTMLDivElement | null>(null);
 
+  const initialFilepath = initialValues?.filepath || '';
```

Comment:
A file path will likely include special characters, so any query params passed to the app likely will need to be URL encoded. So 2 things:
1. Ensure all query params are URL encoded from the source JupyterLab implementation
2. Ensure all query params are URL decoded here. Something like `decodeURIComponent`

Reply:
@jbouder For 1., please see https://github.com/nebari-dev/jupyterlab-jhub-apps/blob/0e0faf6ba6d7107fe44e4518c5ac90641bb6bf4b/src/index.ts#L54-L58.

I believe that I've addressed 2. in the latest commit.

Reply:
Side note…if this should be loaded into an iframe, you should be able to add the following query string param now to make the form load “headless”: ‘headless=true’